merlion/models/forecast/trees.py

```diff
@@ -110,70 +94,62 @@ def _default_train_config(self):
-            if self.prediction_stride > 1:
-                max_forecast_steps = seq_ar_common.max_feasible_forecast_steps(train_data, self.maxlags)
-                if self.max_forecast_steps is not None and self.max_forecast_steps > max_forecast_steps:
-                    logger.warning(
-                        f"With train data of length {len(train_data)} and "
-                        f"maxlags={self.maxlags}, the maximum supported forecast "
-                        f"steps is {max_forecast_steps}, but got "
-                        f"max_forecast_steps={self.max_forecast_steps}. Reducing "
-                        f"to the maximum permissible value or switch to "
-                        f"'training_mode = autogression'."
-                    )
-                    self.config.max_forecast_steps = max_forecast_steps
-                if self.max_forecast_steps is not None and self.prediction_stride != self.max_forecast_steps:
-                    logger.warning(
-                        f"For multivariate dataset, reset prediction_stride = max_forecast_steps = {self.max_forecast_steps} "
-                    )
-                    self.config.prediction_stride = self.max_forecast_steps
-                # process train data
-                inputs_train, labels_train, labels_train_ts = seq_ar_common.process_rolling_train_data(
-                    train_data, self.target_seq_index, self.maxlags, self.prediction_stride, self.sampling_mode
-                )
-                if fit:
-                    self.model.fit(inputs_train, labels_train)
-                # since the model may predict multiple steps, we concatenate all the first steps together
-                pred = self.model.predict(np.atleast_2d(inputs_train))[:, 0].reshape(-1)
-            else:
-                # autoregression mode
-                prior_forecast, labels_train_ts, pred = self.autoregression_train(
-                    data=train_data, maxlags=self.maxlags, sampling_mode=self.sampling_mode, fit=fit
-                )
+            self.config.prediction_stride = self.max_forecast_steps
+
+        # process train data to the rolling window dataset
+        dataset = RollingWindowDataset(data=train_data,
+                                       target_seq_index=self.target_seq_index,
+                                       maxlags=self.maxlags,
+                                       forecast_steps=self.prediction_stride,
+                                       batch_size=None,
+                                       label_axis=0,
+                                       )
+        inputs_train, labels_train, labels_train_ts = next(iter(dataset))
+
+        if fit:
+            self.model.fit(inputs_train, labels_train)
+        # since the model may predict multiple steps, we concatenate all the first steps together
+        pred = self.model.predict(np.atleast_2d(inputs_train))[:, 0].reshape(-1)
```

Comment:
It seems like this code is duplicated from the base class. Why is this? I feel like the rolling window dataset should still be valid for training.

Reply:
the sampling dimension is different, inside its parent class the rolling window is along the sequence dim, in this case, it is along the time dimension